src/libtiled/tilelayer.h

```diff
@@ -568,6 +571,19 @@ inline void TileLayer::setSize(const QSize &size)
     mHeight = size.height();
 }
 
+inline QRect TileLayer::bounds() const
+{
+    QRect bound = mBounds.boundingRect();
+
+    int x = std::min(0, bound.x());
+    int y = std::min(0, bound.y());
+    int width = std::max(mWidth, bound.width());
+    int height = std::max(mHeight, bound.height());
+
+    return QRect(x, y, width, height).translated(mX, mY);
```

Comment:
I don't see `TileLayer::bounds` being used by `BrushItem`. If you do use it there and it wants the bounds to be translated, then yes you should either leave this in there or add a translation in `BrushItem`.